engine/core.py

```diff
@@ -106,11 +106,19 @@ def start_game(self) -> None:
 
         self.current_state = self.ingame_state
 
-    def change_region(self, name: str, start_location: str) -> None:
-        """Changes the region of the game."""
-        if self.world is None:
+    def world_exist(self):
```

Comment:
Unfortunately pulling these out into methods will mean mypy is no longer able to do [type narrowing](https://mypy.readthedocs.io/en/stable/type_narrowing.html), so the CI checks will fail. You could potentially use type guards for this, but that's probably a pain and not worth the effort.